src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -73,7 +78,14 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             LogicalPlanFingerprint.Properties(Seq(Signature(signatureProvider.name, s)))))
 
         val coveringIndexProperties =
-          (hasLineageProperty(spark) ++ hasParquetAsSourceFormatProperty(relation)).toMap
+          (hasLineageProperty(spark) ++
+            hasParquetAsSourceFormatProperty(relation)).toMap ++
+            Hyperspace
+              .getContext(spark)
+              .sourceProviderManager
+              .enrichIndexProperties(
+                sourcePlanProperties.relations.head,
+                prevIndexProperties() + (IndexConstants.INDEX_LOG_VERSION -> versionId.toString))
```

Comment:
Can we pack lineage / parquet property here as well? The intention is to pack all the properties first then pass them to `enrichIndexProperties` to "enrich" them as needed by the provider.